src/solc_0.8/defi/rules-v2/ContributionRulesV2.sol

```diff
@@ -0,0 +1,341 @@
+//SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.2;
+
+import {Ownable} from "@openzeppelin/contracts-0.8/access/Ownable.sol";
+import {Address} from "@openzeppelin/contracts-0.8/utils/Address.sol";
+import {SafeMathWithRequire} from "../../common/Libraries/SafeMathWithRequire.sol";
+import {IERC721} from "@openzeppelin/contracts-0.8/token/ERC721/IERC721.sol";
+import {IERC1155} from "@openzeppelin/contracts-0.8/token/ERC1155/IERC1155.sol";
+import {IContributionRules} from "../interfaces/IContributionRules.sol";
+
+/// @notice The contribution rules is a plugin contract that takes the amount staked and the address of
+/// the user and returns the absolute share (contribution) that the user will get from the total rewards.
+/// The contribution rules must implement the IContributionRules interface. This interface has only one method:
+/// computeMultiplier which takes the account and amountStaked parameters and returns the contribution for this user.
+contract ContributionRulesV2 is Ownable, IContributionRules {
```

Comment:
since we have a folder rules-v2, do we want to rename ContributionRules to ContributionRulesV2 ? (let me guess, because of the deploy script ?)

Reply:
Yes, because of the deploy script and deployment file.